single_kernel_mongo/state/charm_state.py

```diff
@@ -537,8 +553,10 @@ def operator_config(self) -> MongoConfiguration:
     @property
     def mongos_config(self) -> MongoConfiguration:
         """Mongos Configuration for the mongos user."""
-        username = self.secrets.get_for_key(Scope.APP, key=AppPeerDataKeys.username.value)
-        password = self.secrets.get_for_key(Scope.APP, key=AppPeerDataKeys.password.value)
+        if self.charm_role.name == KindEnum.MONGOD:
+            return self.mongos_config_for_user(OperatorUser, self.app_hosts)
```

Comment:
if the kind above is mongod, do we want to set mongos?

Reply:
Because we're getting the mongoS configuration for the mongoDB charm which is useful when we run in a sharded deployment context

Reply:
Ah so `if self.charm_role.name == KindEnum.MONGOD:` is a check of "is this charmed MongoDB or Charmed Mongos"? If so maybe the kind should be renamed to `charmed_mongodb` otherwise it seems like the daemon

Reply:
Mehdi made a similar comment about the name `KindEnum` in another PR.
I have to come up with a better naming.
The `charmed_mongodb` brings a good idea and I'll could go with
```python3
class CharmKind(str, Enum):
    MONGODB = "mongodb"
    MONGOS = "mongos"
```
Does it make more sense?

Reply:
omg I would love that approach! Thanks for being flexible! 🤩

Reply:
See b31de6b